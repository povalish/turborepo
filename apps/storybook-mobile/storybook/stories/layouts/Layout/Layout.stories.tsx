import React from 'react';

import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';

import { Layout, LayoutText } from 'ui-kit/layouts/Layout/Layout.native';
import { Wrapper } from '../Wrapper';

// Layout.stories
//

storiesOf('Layout', module)
  .addDecorator((getStory) => <Wrapper>{getStory()}</Wrapper>)
  .add('Default', () => (
    <Layout>
      <LayoutText>{text('Button text', 'Hello Button')}</LayoutText>
    </Layout>
  ));
