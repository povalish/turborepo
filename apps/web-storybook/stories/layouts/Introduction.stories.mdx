<Meta title="Layouts/Introduction" />

# UI Layouts

Storybook for Web UI Library
