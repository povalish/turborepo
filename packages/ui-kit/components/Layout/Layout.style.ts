import { cva } from 'class-variance-authority';

export const styles = cva(`
  w-screen h-screen 
  bg-[#6FD0F3] text-[#333E53]
  flex flex-col items-center justify-center
`);
