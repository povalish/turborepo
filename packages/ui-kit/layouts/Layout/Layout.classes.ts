import { cva } from 'class-variance-authority';

// Layout.classes.ts
//

export const containerClasses = cva(`
  w-screen h-screen 
  bg-[#6FD0F3]
  flex flex-col items-center justify-center
  px-4
`);

export const textClasses = cva(`
  text-[#333E53] text-3xl text-center
  font-bold
`);

export const classes = cva(`${containerClasses()} ${textClasses()}`);
