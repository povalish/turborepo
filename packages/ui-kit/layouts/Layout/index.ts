export * from './Layout';
export * from './Layout.classes';
