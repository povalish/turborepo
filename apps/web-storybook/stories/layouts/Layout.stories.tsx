import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Layout } from 'ui-kit/layouts/Layout';

// Button.stories
//

export default {
  title: 'Layouts/Layout',
  component: Layout,
} as ComponentMeta<typeof Layout>;

const Template: ComponentStory<typeof Layout> = (args) => <Layout {...args}>Layout Wrapper</Layout>;

export const DefaultLayout = Template.bind({});
