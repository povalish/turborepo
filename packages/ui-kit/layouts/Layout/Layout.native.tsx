import React from 'react';
import { ViewProps } from 'react-native';
import { StyledProps } from 'nativewind';
import { LayoutView } from './Layout.style';

// Layout
//
interface IView extends StyledProps<ViewProps> {}

export const Layout: React.FC<IView> = ({ children, ...props }) => (
  <LayoutView {...props}>{children}</LayoutView>
);
