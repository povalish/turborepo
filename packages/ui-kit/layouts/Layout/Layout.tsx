import React, { PropsWithChildren } from 'react';
import { styles } from './Layout.style';

// Layout
//

export const Layout: React.FC<PropsWithChildren> = ({ children }) => (
  <div className={styles()}>{children}</div>
);
