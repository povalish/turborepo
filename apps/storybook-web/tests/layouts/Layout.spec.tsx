import { render } from '@testing-library/react';
import { Layout } from 'ui-kit/layouts/Layout';

describe('ui components > Layout', () => {
  it('renders component unchanged', () => {
    const { container } = render(<Layout>Sample Text</Layout>);
    expect(container).toMatchSnapshot();
  });
});
