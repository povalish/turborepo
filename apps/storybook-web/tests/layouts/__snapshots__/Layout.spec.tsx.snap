// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui components > Layout renders component unchanged 1`] = `
<div>
  <div
    class="
  w-screen h-screen 
  bg-[#6FD0F3]
  flex flex-col items-center justify-center
  px-4
 
  text-[#333E53] text-3xl text-center
  font-bold
"
  >
    Sample Text
  </div>
</div>
`;
